wordpress-mobile/WordPress-Android
# WP Stories integration - save event snackbars

This PR builds on top of #11930 

This PR implements snackbars on the MySiteFragment for:
- start Story saving process
- Story saving process ending successfully
- Story saving process ending in error

This PR uses a similar structure for event posting with EventBus as per the Media and Post uploaders in the `UploadService`, with some analogies for example the `StorySaveEvents` class holds all possible events for the Saving process now (similarly to how `PostEvents` holds events for Post uploading), and hence we can now listen to these in a more structured way.

1. moved start and results events into its own separate class so it's easier to read (`StorySaveEvents`)
2. added the EventBus listeners to MySite fragment as per UploadErrorEvent
3. to handle snackbar through the sequencer, we're using the  `UploadUtilsWrapper` which was already there serving the same purpose. Given both the `FrameSaveService` and `UploadService` are going to be tightly connected (one chaining into the other), makes sense to reuse the same Utils class for snackbars instead of duplicating code unnecessarily.


To test:

Successful path
1. tap on the FAB
2. select Story
3. when the camera preview appears, take a picture
4. optionally, add emoji  / text
5. tap on the `PUBLISH` buttton on the top-right corner of the screen
6. verify the app returns to the previous (main) screen and both he start and end snackbars are shown.

Error path
1. introduce an error in saving (i.e. artificial exception in mp4composerengine such as the one int [this gist](https://gist.github.com/mzorz/da69f0dea9918877defefeb0573b560d))
2. create a video and add some text or emoji
3. tap publish
4. observe the error notification and error snackbar appear
5. tap on MANAGE action on the error snackbar
6. observe the error handling view appears
7. tap RETRY,   see it still fails
8. tap on the Cross to exit the screen, and when prompted about discarding the Story, tap DISCARD
9. observe no snackbar appears (meaning, the sticky event added by the Service has been successfully removed - see commit 5397a25fbeba391f1d795aba429129300d8391fe in Stories repo)
10. observe the error notification (originally created and appearing in step 4 above) is not there anymore in the system dashboard (meaning it has been correctly cleard when discarding the Sotry on step 8 above). (see related commit bbce9a03e53b7c8cb0da3ded5b8da65bdabfd1da in Stories repo, and de7d5c50b2b6e68d203f048c987cabf1ca36cd73 in WPAndroid repo)


PR submission checklist:
